### PR TITLE
Env variable WITHDRAWN_HIJACK_THRESHOLD to consider hijacks withdrawn

### LINKED ISSUE #255: Env variable for withdrawn hijack characterization threshold
**Is your feature request related to a problem? Please describe.**
In some rare cases some monitors may not see all hijack withdrawals or they see simultaneously a withdrawal and an update, preventing the withdrawn characterization of a hijack.

**Describe the solution you'd like**
Define env variable `WITHDRAWN_HIJACK_THRESHOLD` which is effectively used as follows: in database.py instead of checking for strict equality between the A and W peer sets in withdrawals, tag a hijack as withdrawn when `WITHDRAWN_HIJACK_THRESHOLD`*|peers_seen_update|/100.0<=|peers_seen_withdrawal|.

When this is set to 100 percent, the behavior will be exactly as now.

**Describe alternatives you've considered**
N/A

**Additional context**
User will be able to set his/her own thresholds.

### PR DESCRIPTION
**Pull request by vkotronis**
### Description of PR

What component(s) does this PR affect?

- [X] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #255 

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [X] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

Need to update: https://github.com/FORTH-ICS-INSPIRE/artemis/wiki/Environment-variables

### CODE REVIEW

File: backend/docker-compose.test-backend.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ services:
             HIJACK_LOG_FILTER: ${HIJACK_LOG_FILTER}
             MON_TIMEOUT_LAST_BGP_UPDATE: ${MON_TIMEOUT_LAST_BGP_UPDATE}
             HIJACK_LOG_FIELDS: ${HIJACK_LOG_FIELDS}
+            WITHDRAWN_HIJACK_THRESHOLD: 70
```
Comment (vkotronis) [inline backend/docker-compose.test-backend.yml]: I put this here for the test (leave as-is, useful for running automated tests!)

File: backend/core/database.py [hunk 2]
```diff
@@ -1309,7 +1310,10 @@ def _handle_bgp_withdrawals(self):
                         if withdrawal[1] not in entry[1] and withdrawal[1] in entry[0]:
                             entry[1].append(withdrawal[1])
                             timestamp = max(withdrawal[2], entry[6])
-                            if len(entry[0]) == len(entry[1]):
+                            # if a certain percentage of hijack 'A' peers see corresponding hijack 'W'
+                            if len(entry[1]) >= int(
```
Comment (vkotronis) [inline backend/core/database.py]: basic logic, check carefully!

File: backend/docker-compose.test-backend.yml [hunk 3]
```diff
@@ -74,6 +74,7 @@ services:
             HIJACK_LOG_FILTER: ${HIJACK_LOG_FILTER}
             MON_TIMEOUT_LAST_BGP_UPDATE: ${MON_TIMEOUT_LAST_BGP_UPDATE}
             HIJACK_LOG_FIELDS: ${HIJACK_LOG_FIELDS}
+            WITHDRAWN_HIJACK_THRESHOLD: 60
```
Comment (vkotronis) [inline backend/docker-compose.test-backend.yml]: leave as-is for automated tests

Comment (vkotronis) [reply backend/docker-compose.test-backend.yml]: do not change the default env var though!

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **968**
- Hijacks Detected (hijacks/sec): **85**

**Pull request merged**